### PR TITLE
refine weight arg in loss

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR types

Function optimization
### PR changes

APIs 
### Describe

主要修改点：
1. 完善 `Loss` 模块对 `weight` 参数的支持，以及修改对应 docstring

### CODE REVIEW

File: ppsci/loss/l2.py [hunk 1]
```diff
@@ -56,20 +56,28 @@ def forward(self, output_dict, label_dict, weight_dict=None):
                 loss *= output_dict["area"]
 
             loss = loss.sum()
+            if isinstance(self.weight, float):
```
Comment (zhiminzhang0830) [inline ppsci/loss/l2.py:59]: 同上

File: ppsci/loss/mse.py [hunk 2]
```diff
@@ -45,8 +45,8 @@ class MSELoss(base.Loss):
 
     def __init__(
         self,
-        reduction: str = "mean",
-        weight: Optional[Union[Dict[str, float], float]] = None,
+        reduction: Literal["mean", "sum"] = "mean",
```
Comment (zhiminzhang0830) [inline ppsci/loss/mse.py:48]: 更新下docstring

File: ppsci/loss/mse.py [hunk 3]
```diff
@@ -72,6 +72,11 @@ def forward(self, output_dict, label_dict, weight_dict=None):
                 loss = loss.sum()
             elif self.reduction == "mean":
                 loss = loss.mean()
+            if isinstance(self.weight, float):
```
Comment (zhiminzhang0830) [inline ppsci/loss/mse.py:75]: 重复

File: ppsci/loss/mse.py [hunk 4]
```diff
@@ -126,13 +136,16 @@ class PeriodicMSELoss(base.Loss):
         reduction (str, optional): Reduction method. Defaults to "mean".
     """
 
-    def __init__(self, reduction="mean"):
-        super().__init__()
+    def __init__(
+        self,
+        reduction: Literal["mean", "sum"] = "mean",
```
Comment (zhiminzhang0830) [inline ppsci/loss/mse.py:141]: 更新docstring

File: ppsci/loss/integral.py [hunk 5]
```diff
@@ -45,8 +46,8 @@ class IntegralLoss(base.Loss):
 
     def __init__(
         self,
-        reduction: str = "mean",
-        weight: Optional[Union[Dict[str, float], float]] = None,
+        reduction: Literal["mean", "sum"] = "mean",
```
Comment (zhiminzhang0830) [inline ppsci/loss/integral.py:49]: 更新docstring

File: ppsci/loss/integral.py [hunk 6]
```diff
@@ -73,5 +74,11 @@ def forward(self, output_dict, label_dict, weight_dict=None):
                 loss = loss.sum()
             elif self.reduction == "mean":
                 loss = loss.mean()
+
+            if isinstance(self.weight, float):
```
[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply ppsci/loss/integral.py:74]: 已删除

File: ppsci/loss/l2.py (hunk 1 above)
Comment (HydrogenSulfate) [reply ppsci/loss/l2.py:55]: 已删除

File: ppsci/loss/mse.py (hunk 2 above)
Comment (HydrogenSulfate) [reply ppsci/loss/mse.py:48]: 已添加weight

Comment (HydrogenSulfate) [reply ppsci/loss/mse.py:48]: 已更新

File: ppsci/loss/mse.py (hunk 3 above)
Comment (HydrogenSulfate) [reply ppsci/loss/mse.py:71]: 已删除

File: ppsci/loss/mse.py (hunk 4 above)
Comment (HydrogenSulfate) [reply ppsci/loss/mse.py:139]: 已更新

File: ppsci/loss/integral.py (hunk 5 above)
Comment (HydrogenSulfate) [reply ppsci/loss/integral.py:49]: 已更新
